File: src/pycropml/transpiler/antlr_py/tests/examples/SimplaceComponent/Simplace_SoilTemperature/src/java/Simplace_SoilTemperature/SoiltemperatureState.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;
public class SoilTemperatureState
{
    private Double Albedo;
    private Double SnowWaterContent;
    private Double SoilSurfaceTemperature;
    private Integer AgeOfSnow;
    private Double [] pSoilLayerDepth;
    private Double [] SoilTempArray;
    
    public SoilTemperatureState() { }
    
    public SoilTemperatureState(SoilTemperatureState toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.Albedo = toCopy.getAlbedo();
            this.SnowWaterContent = toCopy.getSnowWaterContent();
            this.SoilSurfaceTemperature = toCopy.getSoilSurfaceTemperature();
            this.AgeOfSnow = toCopy.getAgeOfSnow();
            pSoilLayerDepth = new Double[toCopy.getpSoilLayerDepth().length];
        for (int i = 0; i < toCopy.getpSoilLayerDepth().length; i++)
        {
            pSoilLayerDepth[i] = toCopy.getpSoilLayerDepth()[i];
        }
            SoilTempArray = new Double[toCopy.getSoilTempArray().length];
        for (int i = 0; i < toCopy.getSoilTempArray().length; i++)
        {
            SoilTempArray[i] = toCopy.getSoilTempArray()[i];
        }
        }
    }
    public Double getAlbedo()
    { return Albedo; }

    public void setAlbedo(Double _Albedo)
    { this.Albedo= _Albedo; } 
    
    public Double getSnowWaterContent()
    { return SnowWaterContent; }

    public void setSnowWaterContent(Double _SnowWaterContent)
    { this.SnowWaterContent= _SnowWaterContent; } 
    
    public Double getSoilSurfaceTemperature()
    { return SoilSurfaceTemperature; }

    public void setSoilSurfaceTemperature(Double _SoilSurfaceTemperature)
    { this.SoilSurfaceTemperature= _SoilSurfaceTemperature; } 
    
    public Integer getAgeOfSnow()
    { return AgeOfSnow; }

    public void setAgeOfSnow(Integer _AgeOfSnow)
    { this.AgeOfSnow= _AgeOfSnow; } 
    
    public Double [] getpSoilLayerDepth()
    { return pSoilLayerDepth; }

    public void setpSoilLayerDepth(Double [] _pSoilLayerDepth)
    { this.pSoilLayerDepth= _pSoilLayerDepth; } 
    
    public Double [] getSoilTempArray()
    { return SoilTempArray; }

    public void setSoilTempArray(Double [] _SoilTempArray)
    { this.SoilTempArray= _SoilTempArray; } 
    
}

File: src/pycropml/transpiler/antlr_py/tests/examples/SimplaceComponent/Simplace_SoilTemperature/src/cs/Simplace_SoilTemperature/SoilTemperatureState.cs
using System;
using System.Collections.Generic;
public class SoilTemperatureState 
{
    private double _Albedo;
    private double _SnowWaterContent;
    private double _SoilSurfaceTemperature;
    private int _AgeOfSnow;
    private double[] _pSoilLayerDepth;
    private double[] _SoilTempArray;
    
    public SoilTemperatureState() { }
    
    
    public SoilTemperatureState(SoilTemperatureState toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    _Albedo = toCopy._Albedo;
    _SnowWaterContent = toCopy._SnowWaterContent;
    _SoilSurfaceTemperature = toCopy._SoilSurfaceTemperature;
    _AgeOfSnow = toCopy._AgeOfSnow;
    pSoilLayerDepth = new double[toCopy._pSoilLayerDepth.Length];
            for (int i = 0; i < toCopy._pSoilLayerDepth.Length; i++)
            { _pSoilLayerDepth[i] = toCopy._pSoilLayerDepth[i]; }
    
    SoilTempArray = new double[toCopy._SoilTempArray.Length];
            for (int i = 0; i < toCopy._SoilTempArray.Length; i++)
            { _SoilTempArray[i] = toCopy._SoilTempArray[i]; }
    
    }
    }
    public double Albedo
        {
            get { return this._Albedo; }
            set { this._Albedo= value; } 
        }
    public double SnowWaterContent
        {
            get { return this._SnowWaterContent; }
            set { this._SnowWaterContent= value; } 
        }
    public double SoilSurfaceTemperature
        {
            get { return this._SoilSurfaceTemperature; }
            set { this._SoilSurfaceTemperature= value; } 
        }
    public int AgeOfSnow
        {
            get { return this._AgeOfSnow; }
            set { this._AgeOfSnow= value; } 
        }
    public double[] pSoilLayerDepth
        {
            get { return this._pSoilLayerDepth; }
            set { this._pSoilLayerDepth= value; } 
        }
    public double[] SoilTempArray
        {
            get { return this._SoilTempArray; }
            set { this._SoilTempArray= value; } 
        }
}